l2geth/rollup/sync_service.go

```diff
@@ -710,13 +760,49 @@ func (s *SyncService) applyTransaction(tx *types.Transaction) error {
 	return nil
 }
 
+// verifyFee will verify that a valid fee is being paid.
+func (s *SyncService) verifyFee(tx *types.Transaction) error {
+	// Exit early if fees are enforced and the gasPrice is set to 0
+	if s.enforceFees && tx.GasPrice().Cmp(common.Big0) == 0 {
+		return errors.New("cannot accept 0 gas price transaction")
+	}
+
+	l1GasPrice, err := s.RollupGpo.SuggestDataPrice(context.Background())
+	if err != nil {
+		return err
+	}
+	l2GasPrice, err := s.RollupGpo.SuggestExecutionPrice(context.Background())
+	if err != nil {
+		return err
+	}
+	// Calculate the fee based on decoded L2 gas limit
+	gas := new(big.Int).SetUint64(tx.Gas())
+	l2GasLimit := core.DecodeL2GasLimit(gas)
+	fee, err := core.CalculateRollupFee(tx.Data(), l1GasPrice, l2GasLimit, l2GasPrice)
+	if err != nil {
+		return err
+	}
+	// If fees are not enforced and the gas price is 0, return early
+	if !s.enforceFees && tx.GasPrice().Cmp(common.Big0) == 0 {
+		return nil
+	}
+	// Make sure that the fee is paid
+	if tx.Gas() < fee.Uint64() {
```

Comment:
Analyze volatile fees on L1 causing this to fail

Reply:
are these like reminders you set to yourself?